### PR TITLE
feat(InstantSearch.js): move to v2

### PR DESCRIPTION
**Pull request by vvo**
Move to IS.js v2 along with adding some safe guards on running the env.

Linked to https://github.com/algolia/doc/pull/1103

### CODE REVIEW

File: search.js [hunk 1]
```diff
@@ -3,7 +3,20 @@
...
   appId: 'latency',
   apiKey: '6be0576ff61c053d5f9a3225e2a90f76',
-  indexName: 'instant_search'
+  indexName: 'instant_search',
+  searchParameters: {
+    hitsPerPage: 10,
+  },
+  searchFunction(helper) {
+    if (helper.state.query.length > 0) {
+      document.querySelector('.ais-search-box--magnifier').style.display =
```
Comment (bobylito) [inline search.js]: I'm not sure I understand the motivation of this implementation?

**Comment by bobylito** [comment]
Can you deploy the example on surge?

File: search.js (hunk 1 above)
Comment (vvo) [reply search.js]: On https://preview.algolia.com/instantsearch/, the searchbox looks like this:

image

The magnifying glass is then replaced with the cross when the query is not empty.

**Comment by vvo** [comment]
cc @bobylito Just added deploy previews :)

https://deploy-preview-27--instantsearch-demo.netlify.com

**Comment by bobylito** [comment]
❤️

**Comment by bobylito** [comment]
Can we merge? :)

**Pull request merged**